spec/models/miq_websocket_worker/runner_spec.rb

```diff
@@ -0,0 +1,23 @@
+describe MiqWebsocketWorker::Runner do
+  describe '#check_internal_thread' do
+    subject do
+      w = described_class.allocate
+      allow(w).to receive(:worker_initialization)
+      w.send(:initialize, :guid => MiqWebsocketWorker.create_worker_record.guid)
+      w
+    end
+
+    let(:worker) { double }
+    let(:app) { WebsocketServer.new }
+
+    it 'exits if the thread is not running' do
+      subject.instance_variable_set(:@worker, worker)
+      allow(worker).to receive(:rails_application).and_return(app)
+      expect(subject).to receive(:do_exit)
+
+      app.instance_variable_get(:@transmitter).kill
+      sleep(1) # Give some time to the thread to stop
```

Comment:
Can you do this instead of sleeping a whole second?

```ruby
sleep 0.01 until !app.instance_variable_get(:@transmitter).alive?
```

http://ruby-doc.org/core-2.2.0/Thread.html#method-i-alive-3F